### PR TITLE
Fix the potential redundant_clone issue of path

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: Aisuko <urakiny@gmail.com>

Hi @PhilipK , thanks for your work. 

It is awesome. I fix a little bit potential issue for BoileR. I'm a big fan of Steam Deck, although my order was suspended in Canada.

Even though I cannot test it on my Steam Deck, I add a simple test case to support it does not change any useful feature. Here below,

```
 *  Executing task: cargo test --package boilr --bin boilr -- config::tests::check_return_config_format --exact --nocapture 

   Compiling boilr v1.5.1 (/workspaces/BoilR)
    Finished test [unoptimized + debuginfo] target(s) in 28.74s
     Running unittests src/main.rs (target/debug/deps/boilr-23166f7cd679110a)

running 1 test
test config::tests::check_return_config_format ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 24 filtered out; finished in 0.00s

 *  Terminal will be reused by tasks, press any key to close it. 
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -49,5 +49,22 @@ pub fn get_backups_flder() -> PathBuf {
...
+#[cfg(test)]
+mod tests{
+    use std::path::PathBuf;
+
+    use super::get_config_folder;
+
+    #[test]
+    fn check_return_config_format(){
+        let path=get_config_folder();
+        let val=std::env::var("HOME").unwrap();
```
Comment (PhilipK) [inline src/config.rs:66]: This will always fail on Windows, since there is no HOME variable.

File: src/config.rs [hunk 2]
```diff
@@ -49,5 +49,22 @@ pub fn get_backups_flder() -> PathBuf {
 
 #[cfg(target_family = "unix")]
 pub fn get_boilr_links_path() -> PathBuf {
-    get_config_folder().join("links").to_path_buf()
+    // Remove redundant_clone. Refer: https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
```
Comment (PhilipK) [inline src/config.rs:52]: No reason to have a comment that something was removed. 
Its a good change, but no need to add a comment.

```suggestion
```

File: src/config.rs [hunk 3]
```diff
@@ -49,5 +49,22 @@ pub fn get_backups_flder() -> PathBuf {
...
+
+
+#[cfg(test)]
+mod tests{
+    use std::path::PathBuf;
+
+    use super::get_config_folder;
+
+    #[test]
+    fn check_return_config_format(){
```
Comment (PhilipK) [inline src/config.rs:64]:
```suggestion
    #[test]
    #[cfg(target_family = "unix")]
    fn check_return_config_format(){
```

File: src/config.rs (hunk 1 above)
Comment (PhilipK) [inline src/config.rs:66]: The test need to handle that "XDG_CONFIG_HOME" will be used if it is defined, and then HOME will be used otherwise.

Comment (Aisuko) [reply src/config.rs:66]: I'd like to change this test case just cover the Unix environment.

File: src/config.rs (hunk 3 above)
Comment (Aisuko) [reply src/config.rs]: Done

File: src/config.rs (hunk 1 above)
Comment (Aisuko) [reply src/config.rs]: Done.

**Comment by Aisuko** [comment]
Thank you quick reply. Please take a look.

File: src/config.rs [hunk 4]
```diff
@@ -49,5 +49,34 @@ pub fn get_backups_flder() -> PathBuf {
...
+    use std::path::PathBuf;
+
+    use super::get_config_folder;
+
+    #[test]
+
+    #[cfg(target_family = "unix")]
+    fn check_return_config_format(){
+        // Parameters for set environment 'XDG_CONFIG_HOME'
+        // std::env::set_var("XDG_CONFIG_HOME", std::env::var("HOME").unwrap()+"/.config/xdg");
```
Comment (PhilipK) [inline src/config.rs:67]: Maybe turn this into two tests, instead of having some commented out code?

Comment (Aisuko) [reply src/config.rs:67]: Good idea. I will update later.

Comment (Aisuko) [reply src/config.rs]:
```
 *  Executing task: cargo test --package boilr --bin boilr -- config::tests --nocapture 

   Compiling boilr v1.5.1 (/workspaces/BoilR)
    Finished test [unoptimized + debuginfo] target(s) in 33.18s
     Running unittests src/main.rs (target/debug/deps/boilr-23166f7cd679110a)

running 2 tests
test config::tests::check_return_xdg_config_path ... ok
test config::tests::check_return_config_path ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 24 filtered out; finished in 0.00s
```

**Comment by Aisuko** [comment]
Please take a look.

**Comment by Aisuko** [comment]
I have no idea the reason it failed on Ubuntu and Flatpak. Please help me with this. Thanks.

**Comment by Aisuko** [comment]
Hi @PhilipK , feel free to take a look.

**Review by PhilipK — COMMENTED** [review]
I think the problem is that you need to add a /boilr

File: src/config.rs [hunk 5]
```diff
@@ -49,5 +49,42 @@ pub fn get_backups_flder() -> PathBuf {
...
+
+    use super::get_config_folder;
+
+    #[test]
+
+    #[cfg(target_family = "unix")]
+    fn check_return_xdg_config_path(){
+
+        // Parameters for set environment 'XDG_CONFIG_HOME'
+        std::env::set_var("XDG_CONFIG_HOME", std::env::var("HOME").unwrap()+"/.config/xdg");
```
[the remark this answers is not in the document]
Comment (Aisuko) [reply src/config.rs]: Thank you. It works for me now.  Feel free to merge it. Thanks

**Comment by PhilipK** [comment]
The unit test still seems to be failing

**Comment by Aisuko** [comment]
> The unit test still seems to be failing

I see. I updated the code.

**Pull request merged**